src/librustc_typeck/check/method/suggest.rs

```diff
@@ -771,7 +774,28 @@ fn compute_all_traits<'a, 'gcx, 'tcx>(tcx: TyCtxt<'a, 'gcx, 'tcx>) -> Vec<DefId>
                 _ => {}
             }
         }
+
+        let mut cnums: FxHashSet<CrateNum> = FxHashSet::default();
+
+        // Attempt to load all crates that we have `--extern` flags for, this means
+        // we will be able to make suggestions for traits they define. Particularly useful
+        // in Rust 2018 as there aren't `extern crate` lines that import a crate even if it isn't
+        // otherwise used.
+        for (name, _) in tcx.extern_prelude.clone() {
```

Comment:
Fixed, thanks.

Reply:
You shouldn't need to clone anything here, just call `.keys()` on it.